test/conformance/runtime/readiness_probe_test.go

```diff
@@ -91,6 +99,30 @@ func TestProbeRuntime(t *testing.T) {
 				if err != nil {
 					t.Fatalf("Failed to create initial Service: %v: %v", names.Service, err)
 				}
+
+				// Once the service reports ready we should immediately be able to curl it.
+				client, err := pkgtest.NewSpoofingClient(context.Background(), clients.KubeClient, t.Logf, resources.Route.Status.URL.URL().Host,
+					test.ServingFlags.ResolvableDomain, test.AddRootCAtoTransport(context.Background(), t.Logf, clients, test.ServingFlags.HTTPS))
+				if err != nil {
+					t.Fatal("Create spoofing client:", err)
+				}
+
+				url := resources.Route.Status.URL.URL()
+				url.Path = "/healthz"
+				req, err := http.NewRequest(http.MethodGet, url.String(), nil)
+				if err != nil {
+					t.Fatal("NewRequest:", err)
+				}
+
+				resp, err := client.Do(req)
+				if err != nil {
+					t.Fatal("GET ready service:", err)
+				}
+
+				if want, got := test.HelloWorldText, string(resp.Body); want != got {
+					t.Fatalf("Expected body to be %q but was %q:", want, got)
+				}
```

Comment:
Can we make this resilient against Ingress inconsistency (memo to self: bring this up to the networking group again)? This will intermittently fail for ingresses, that can't guarantee consistency through probing (i.e. DNS based front-doors). Using the obligatory `pkgtest.WaitForEndpointState` combined with a strict body check should yield the same result.

Reply:
hm.. yeah.. darn.. I actually intentionally removed waitforendpointstate because I don't want to this to be able to pass by waiting for the thing to go healthy iyswim. I guess as long as the startup delay is higher than the waitforendpointstate polling time it's ok :/.

Reply:
If you use `MatchesBody`, it'll break immediately and won't retry. Only `EventuallyMatchesBody` allows for other responses. Crucially, if it's wrapped with `RetryingRouteInconsistency` that gives downstreams a hook to adjust the test to narrowly retry things without diluting the results. So for your use-case, the semantics should be the same.

Reply:
TIL! let's give that a go

Reply:
> This will intermittently fail for ingresses, that can't guarantee consistency through probing (i.e. DNS based front-doors). 

Can you clarify this I'm confused. I thought net- implementations could set any target for probing - or does it require knowing all the IPs and overfitted to probing pods?

Reply:
@dprotaso see https://github.com/knative/serving/issues/11404